keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -323,12 +323,23 @@ class ApplicationService(
   ): ArtifactVersionSummary {
     val artifactSupplier = artifactSuppliers.supporting(artifact.type)
     val publishedArtifact = artifact.toSpinnakerArtifact(version)
+    val artifactStatus = artifactSupplier.getReleaseStatus(publishedArtifact)
+    val buildMetadata = repository.getArtifactBuildMetadata(artifact.name, artifact.type, version, artifactStatus)
+    val gitMetadata = repository.getArtifactGitMetadata(artifact.name, artifact.type, version, artifactStatus)
```

Comment:
We could simplify this a bit by having a single method `repository.getArtifactMetadata()` and unpacking the build and git metadata here instead of in all repository implementations.